common/persistence/nosql/nosql_execution_store_test.go

```diff
@@ -860,6 +860,211 @@ func TestNosqlExecutionStore(t *testing.T) {
+				// as the operation should not proceed due to validation failure
+				return newTestNosqlExecutionStore(nosqlplugin.NewMockDB(ctrl), log.NewNoop())
+			},
+			testFunc: func(store *nosqlExecutionStore) error {
+				taskInfo := persistence.InternalReplicationTaskInfo{} // Intentionally invalid/incomplete task info
+				return store.PutReplicationTaskToDLQ(ctx, &persistence.InternalPutReplicationTaskToDLQRequest{
+					SourceClusterName: "sourceCluster",
+					TaskInfo:          &taskInfo,
+				})
+			},
+			expectedError: &types.BadRequestError{Message: "Invalid replication task info: TaskID is required"},
+		},
+
+		{
+			name: "GetReplicationTasksFromDLQ success",
+			setupMock: func(ctrl *gomock.Controller) *nosqlExecutionStore {
+				mockDB := nosqlplugin.NewMockDB(ctrl)
+
+				nextPageToken := []byte("next-page-token")
+				replicationTasks := []*persistence.InternalReplicationTaskInfo{}
+				mockDB.EXPECT().
+					SelectReplicationDLQTasksOrderByTaskID(
+						ctx,
+						shardID,
+						"sourceCluster",
+						10,
+						gomock.Any(),
+						int64(0),
+						int64(100),
+					).Return(replicationTasks, nextPageToken, nil)
+				return newTestNosqlExecutionStore(mockDB, log.NewNoop())
+			},
+			testFunc: func(store *nosqlExecutionStore) error {
+				initialNextPageToken := []byte{}
+				_, err := store.GetReplicationTasksFromDLQ(ctx, &persistence.GetReplicationTasksFromDLQRequest{
+					SourceClusterName: "sourceCluster",
+					GetReplicationTasksRequest: persistence.GetReplicationTasksRequest{
+						BatchSize:     10,
+						NextPageToken: initialNextPageToken,
+						ReadLevel:     0,
+						MaxReadLevel:  100,
+					},
+				})
+
+				return err
+			},
+			expectedError: nil,
+		},
+		{
+			name: "GetReplicationTasksFromDLQ failure - invalid read levels",
+			setupMock: func(ctrl *gomock.Controller) *nosqlExecutionStore {
+				return newTestNosqlExecutionStore(nosqlplugin.NewMockDB(ctrl), log.NewNoop())
+			},
+			testFunc: func(store *nosqlExecutionStore) error {
+				_, err := store.GetReplicationTasksFromDLQ(ctx, &persistence.GetReplicationTasksFromDLQRequest{
+					SourceClusterName: "sourceCluster",
+					GetReplicationTasksRequest: persistence.GetReplicationTasksRequest{
+						ReadLevel:     100,
+						MaxReadLevel:  50,
+						BatchSize:     10,
+						NextPageToken: []byte{},
+					},
+				})
+				return err
+			},
+			expectedError: &types.BadRequestError{Message: "ReadLevel cannot be higher than MaxReadLevel"},
+		},
+		{
+			name: "GetReplicationDLQSize success",
+			setupMock: func(ctrl *gomock.Controller) *nosqlExecutionStore {
+				mockDB := nosqlplugin.NewMockDB(ctrl)
+				mockDB.EXPECT().
+					SelectReplicationDLQTasksCount(ctx, shardID, "sourceCluster").
+					Return(int64(42), nil)
+				return newTestNosqlExecutionStore(mockDB, log.NewNoop())
+			},
+			testFunc: func(store *nosqlExecutionStore) error {
+				resp, err := store.GetReplicationDLQSize(ctx, &persistence.GetReplicationDLQSizeRequest{
```

Comment:
It's fine here since you already wrote this but going forward I recommend having separate top level TestXyz function for each store function and avoid a `testFunc` in the table. Table tests should provide inputs, mock setup and expected results. Glueing it all together and doing validations should be left to test body inside t.Run. All table tests share same test body function.

Reply:
got it!